ls1intum/Artemis
# `Development`: Separate checkout tasks from build script for local continuous integration

### Checklist
#### General


#### Server


### Motivation and Context
This PR separates checkout tasks from the build script that is executed within a build job container for the local CI system. This facilitates a cleaner script and is the basis to later facilitate custom created build scripts.

### Description
Cleaned up createBuildScript() and removed redundant copy steps to populate the container with repositories.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- TS 3 or 6

1. As Instructor, create a programming exercise.
2. Verify build results for template and solution
3. As student, create submission for programming exercise
4. Verify build results

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
